### ISSUE TITLE
CellProfiler 2.1.1, how to install wxPython 2.8 instead of wxPython 3.0 in Ubuntu? 

### ISSUE THREAD

**Issue by xueshirong** [CellProfiler/CellProfiler#1978]
Hello,

I am new in CellProfiler. In Ubuntu, I tried to run the CellProfiler 2.1.1. However the "input modules" can't be displayed in the GUI and there also pop up some error tips when I move mouse. As I know in the previous issue I post here(#1811), I tried to install wxPython 2.8 instead of wxPython 3.0.

The problem is I can't find the wxPython 2.8 package I can install through "conda install" or "apt -get install". Do you mind give me some suggestions on how to install wxPython 2.8 directly through package without build the source code of wxPython?

Thank you.

**Comment by 0x00B1** [comment]
```sh
$ sudo apt-get install python-wxgtk2.8
```

**Comment by xueshirong** [comment]
Still does not work.
![screenshot from 2016-04-22 10 46 46](https://cloud.githubusercontent.com/assets/3871607/14746921/7509fbf8-08a1-11e6-95b0-0292153f581c.png)
![screenshot from 2016-04-22 10 47 13](https://cloud.githubusercontent.com/assets/3871607/14746922/750ae270-08a1-11e6-9902-03944d7c7aff.png)

**Comment by 0x00B1** [comment]
Jeeze, man. `sudo apt-get autoremove`!

Are system packages compatible with Anaconda? If not, you will need to use Anaconda to install wxPython.

**Comment by xueshirong** [comment]
My Ubuntu collapsed. I will restore my system, then try again like you said.
Please leave my issue for some days. Thank you so much.

**Comment by xueshirong** [comment]
Hi, I just reinstall my Ubuntu system.
Now, my system packages compatible with Anaconda. Which command should I use to install wxpython 2.8? I use conda install wxpython, it will install wxpython 3.0.

**Comment by 0x00B1** [comment]
wxPython 3.0 works well.

```sh
conda install wxpython
```

**Comment by xueshirong** [comment]
Thank you for your reply. For now, I install wxPython 3.0, the GUI looks
like as below. The "inputs modules" can't be displayed. The same situation
was happened on my Window system, I changed to wxPython 2.8, and works
well. That is why I want to change to wxPython 2.8 on my Ubuntu 14.04.
[image: Inline image 2]

**Comment by dlogan** [comment]
We can't see the image.  But did you try, extrapolating from http://conda.pydata.org/docs/faq.html

```
conda install wxpython=2.8
```
?

**Comment by 0x00B1** [comment]
Maybe there’s a wxGTK and wxPython mismatch?

**Comment by xueshirong** [comment]
If I use wxpython 3.0, the picture is as follow.
If I use "conda install wxpython", it will install wxpython 3.0.
I tried "conda install wxpython=2.8" or "pip install wxpython=2.8", but failed. It shows that there are no match packages.

![qq 20160428203816](https://cloud.githubusercontent.com/assets/3871607/14923679/1fb04442-0e04-11e6-9b63-cccf02c14cb7.png)

**Comment by AnneCarpenter** [comment]
Hi! Still having troubles? If so please re-open the issue and let us know.

**Comment by abulka** [comment]
I agree that `conda install wxpython=2.8` and even `conda install -c krisvanneste wxpython=2.8.12` (see https://anaconda.org/krisvanneste/wxpython) fails due to "No packages found in current osx-64 channels matching: wxpython 2.8.12*".

Is there any way at all to install wxpython 2.8 in miniconda, as I also need this specific version.

**Comment by qqluckiest** [comment]
Hey, 
I just met the same issue. 
Just download the source of wxpython 2.8 and following the introduction of installation which will "replace" the wx3.0.

**Comment by nichollsfj** [comment]
I'm having the same issue on Linux. I'm getting errors when I try to run CellProfiler 2.2, which looks like it could be solved by installing wxpython version 2.8 instead of 3.0 but I'm unable to find a way to install it.
Conda install says it's not available and pip can only find 4.0 or higher

**Comment by bethac07** [comment]
Did you try installing it from source as suggested above?
